### PR TITLE
Configure Servlet 6.1 TCK to run SSL/Certificate tests

### PR DESCRIPTION
**Pull request by arjantijms**

### CODE REVIEW

**Comment by arjantijms** [comment]
Draft: needs release of updated Arquillian connector still

File: appserver/tests/tck/servlet/src/main/java/org/glassfish/servlet/tck/WebArchiveUpdater.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                Archive<?> applicationArchive = description.getArchive();
+
+                if (testClass.getName().contains("ClientCertAnnoTests") && applicationArchive instanceof WebArchive webArchive) {
+                    webArchive
+                    .addAsWebInfResource(
+                        new File("src/main/resources", "sun-web.xml"),
+                        "sun-web.xml")
+                    .addAsLibraries(
+                        Maven.configureResolver()
+                             .workOffline()
```
Comment (avpinchuk) [inline appserver/tests/tck/servlet/src/main/java/org/glassfish/servlet/tck/WebArchiveUpdater.java:65]: Failed when slf4j absent in the local repo

**Comment by avpinchuk** [comment]
Also failed `create-file-user` command if run from projectr root

File: appserver/tests/tck/servlet/src/main/java/org/glassfish/servlet/tck/WebArchiveUpdater.java (hunk 1 above)
Comment (arjantijms) [reply appserver/tests/tck/servlet/src/main/java/org/glassfish/servlet/tck/WebArchiveUpdater.java:65]: > Failed when slf4j absent in the local repo

I duplicated the command the TCK itself was using for other archives. I can remove it there, but the TCK would still fail then.

**Comment by arjantijms** [comment]
> Also failed create-file-user command if run from project root

What command did you exactly used to run it? Do you mean the main GlassFish root? create-file users use the same setup as a couple of other TCKs are using, so they would fail then as well.

**Comment by avpinchuk** [comment]
> > Also failed create-file-user command if run from project root
> 
> What command did you exactly used to run it? Do you mean the main GlassFish root? create-file users use the same setup as a couple of other TCKs are using, so they would fail then as well.

E.g.,

```sh
cd ~/build/glassfish
mvn clean install -Pstaging,tck -pl :jakarta-servlet-tck,:glassfish-external-tck-servlet
```

**Comment by arjantijms** [comment]
> > > Also failed create-file-user command if run from project root
> > 
> > 
> > What command did you exactly used to run it? Do you mean the main GlassFish root? create-file users use the same setup as a couple of other TCKs are using, so they would fail then as well.
> 
> E.g.,
> 
> ```shell
> cd ~/build/glassfish
> mvn clean install -Pstaging,tck -pl :jakarta-servlet-tck,:glassfish-external-tck-servlet
> ```

Thanks, obviously that should work. I'll take a look.

**Comment by avpinchuk** [comment]
Snippet from build log:

```
Executing post boot command using command: [java, -jar, /home/avpinchuk/build/glassfish/appserver/tests/tck/servlet/target/glassfish8/glassfish/modules/admin-cli.jar, create-file-user, --groups, Manager:Employee, --passwordfile, /home/avpinchuk/build/glassfish/javajoe.pass, javajoe, -t]
RemoteCLICommand.prepare throws exception. Invalid file for option: --passwordfile: java.io.FileNotFoundException: /home/avpinchuk/build/glassfish/javajoe.pass (No such file or directory)
Invalid file for option: --passwordfile: java.io.FileNotFoundException: /home/avpinchuk/build/glassfish/javajoe.pass (No such file or directory)
Usage: nadmin [-H|--host <host(default:localhost)>]
	[-p|--port <port(default:4848)>] [-u|--user <user(default:admin)>]
	[-W|--passwordfile <passwordfile>]
	[-t|--terse[=<terse(default:false)>]]
	[-s|--secure[=<secure(default:false)>]]
	[-e|--echo[=<echo(default:false)>]]
	[-I|--interactive[=<interactive(default:true)>]]
	[-?|--help[=<help(default:false)>] 
	[--detach(default:false)] 
	[--notify(default:false)] [subcommand [options] [operands]]
Command create-file-user failed.
```

**Comment by arjantijms** [comment]
> Snippet from build log:
> 
> ```
> Executing post boot command using command: [java, -jar, /home/avpinchuk/build/glassfish/appserver/tests/tck/servlet/target/glassfish8/glassfish/modules/admin-cli.jar, create-file-user, --groups, Manager:Employee, --passwordfile, /home/avpinchuk/build/glassfish/javajoe.pass, javajoe, -t]
> RemoteCLICommand.prepare throws exception. Invalid file for option: --passwordfile: java.io.FileNotFoundException: /home/avpinchuk/build/glassfish/javajoe.pass (No such file or directory)
> Invalid file for option: --passwordfile: java.io.FileNotFoundException: /home/avpinchuk/build/glassfish/javajoe.pass (No such file or directory)
> Usage: nadmin [-H|--host <host(default:localhost)>]
> 	[-p|--port <port(default:4848)>] [-u|--user <user(default:admin)>]
> 	[-W|--passwordfile <passwordfile>]
> 	[-t|--terse[=<terse(default:false)>]]
> 	[-s|--secure[=<secure(default:false)>]]
> 	[-e|--echo[=<echo(default:false)>]]
> 	[-I|--interactive[=<interactive(default:true)>]]
> 	[-?|--help[=<help(default:false)>] 
> 	[--detach(default:false)] 
> 	[--notify(default:false)] [subcommand [options] [operands]]
> Command create-file-user failed.
> ```

Thanks it's fixed now ;)

**Pull request merged**